### PR TITLE
Simplify some places in the compiler where we have esoteric declarations.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/compiler/emitter.ts [hunk 1]
```diff
@@ -2461,8 +2461,8 @@ module ts {
                         return false;
                     case SyntaxKind.LabeledStatement:
                         return (<LabeledStatement>node.parent).label === node;
-                    case SyntaxKind.CatchClause:
-                        return (<CatchClause>node.parent).name === node;
+                    //case SyntaxKind.CatchClause:
+                    //    return (<CatchClause>node.parent).name === node;
```
Comment (JsonFreeman) [inline src/compiler/emitter.ts]: Remove

File: src/compiler/checker.ts [hunk 2]
```diff
@@ -8926,18 +8920,29 @@ module ts {
             var catchClause = node.catchClause;
             if (catchClause) {
                 // Grammar checking
-                if (catchClause.type) {
-                    var sourceFile = getSourceFileOfNode(node);
-                    var colonStart = skipTrivia(sourceFile.text, catchClause.name.end);
-                    grammarErrorAtPos(sourceFile, colonStart, ":".length, Diagnostics.Catch_clause_parameter_cannot_have_a_type_annotation);
+                if (catchClause.variableDeclaration) {
+                    if (catchClause.variableDeclaration.name.kind !== SyntaxKind.Identifier) {
```
Comment (JsonFreeman) [inline src/compiler/checker.ts]: Add test

**Comment by JsonFreeman** [comment]
:+1:

File: tests/baselines/reference/bpSpan_stmts.baseline [hunk 3]
```diff
@@ -217,7 +217,15 @@
     >:=> (line 28, col 8) to (line 28, col 22)
 29 >    } catch (e) {
 
-         ~~~~~~~~~~~~~                            => Pos: (416 to 428)  SpanInfo: {"start":437,"length":15}
+         ~~~~~~~~                                 => Pos: (416 to 423)  SpanInfo: {"start":437,"length":15}
+    >if (obj.z < 10)
+    >:=> (line 30, col 8) to (line 30, col 23)
+29 >    } catch (e) {
+
+                 ~                                => Pos: (424 to 424)  SpanInfo: undefined
```
Comment (mhegazy) [inline tests/baselines/reference/bpSpan_stmts.baseline]: i do not think this is correct. you can not put a breakpoint on "e".

File: src/compiler/checker.ts (hunk 2 above)
Comment (CyrusNajmabadi) [inline src/compiler/checker.ts]: Done.

**Pull request merged**